packages/entities-interactor/CourseInteractor.ts

```diff
@@ -127,6 +127,17 @@ class CourseInteractor extends AbstractCourseInteractor {
     return this.progressStorage.resetProgress(this.courseId);
   }
 
+  public resetChildrenProgress(id: string) {
+    const entity = this.course.findEntity(id);
+
+    if (entity && isInternalNode(entity) && entity.getChildren()) {
+      forEach(child => {
+        this.progress[child.getId()] = { progress: Progress.Unseen };
+        this.resetChildrenProgress(child.getId());
+      }, (entity as InternalNode).getChildren());
```

Comment:
You don't need `entity as InternalNode` here because you have the type guard in the if condition

Reply:
Furthermore, you could remove `entity.getChildren()` from the if condition. If it doesn't return an array all the time, you can do `entity.getChildren() || []` below